src/SingleMovie.js

```diff
@@ -6,13 +6,15 @@ class SingleMovie extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      movie: null
+      movie: null,
+      error: null,
     }
   }
 
   getSingleMovie = (id) => {
     fetchAPI.fetchSingleMovie(id)
```

Comment:
Just based on the function names, it seems like you have two distinct fetches - one for all movies, one for an individual movie. I'm guessing you could refactor these into a single fetch if you had it receive the entire URL as an argument. Let me know if you have any questions about this.